File: PersonnelDepartment/Tools/Results/Result.cs
namespace PersonnelDepartment.Tools.Results;

//TASK ДОДЕЛАТЬ КЛАСС
public class Result
{
    public Boolean IsSuccess => Errors.Length == 0;
    public Error[] Errors { get; set; }

    public Result(params Error[] errors)
    {
        Errors = errors;
    }

    public static Result Success() => new Result();

    public static Result Fail(String message) => new Result(new Error(message));
    public static Result Fail(Error error) => new Result(error);
    public static Result Fail(Error[] errors) => new Result(errors);
}

File: PersonnelDepartment/ClientApp/src/tools/result/result.ts
import { Error } from "../errors/error";

export class Result {
	public isSuccess = this.errors.length === 0;

	constructor(
		public errors: Error[]
	) { }

	public static success(): Result {
		return new Result([]);
	}

	public static failed(errors: Error[]): Result {
		return new Result(errors);
	}
}

export class FailResult {
	public isSuccess: false = false;
	public data: null = null;
	public errors: Error[];

	public get errorsAsString() {
		return this.errors[0].errorMessage;
	}

	constructor(errors: Error[]) {
		this.errors = errors;
	}
}

export class SuccessResult {
	public isSuccess: true = true;
	public data: null = null;
	public errors: Error[] = [];
}